docs/references/architecture/adr-x-intelligent-validator-peering.md

```diff
@@ -0,0 +1,78 @@
+
+# ADR X: Intelligent Validator Peering
+
+## Changelog
+
+- 11/8: Initial draft
+
+## Status
+
+## Context
+
+The existing CometBFT p2p network is agnostic to the distinction between full nodes and validators. Peers are added via config files and PEX, but validators are equally likely to connect to full nodes as they are to validators via PEX. Thus, validators can be peered suboptimally, requiring messages go through extra hops through full nodes in order for the network to proceed through consensus.
+
+This document proposes a way for validators to peer intelligently with each other, increasing the tendency for validators to be directly peered and thus for consensus messages to travel through the network of validators more quickly. We propose adding a reactor for validator p2p, similar to pex, but specifically for the purpose of getting validators peered with each other.
+
+Note that in this document we will refer to validators as any of the core nodes operated by validator entities in the network. They can be sentry nodes or any non-signing nodes, but should be on the hot-path for consensus messages (i.e. they must participate in the network in order for some consensus messages to be propagated).
+
+## Alternative Approaches
+
+- One alternative for the reactor implementation is complete validator peering, where all validators are peered with all other validators. This is not chosen because it can become prohibitively expensive to a large number of peers, when the number of validators is large.
+
+
+## Decision
+
+## Detailed Design
+
+The CometBFT config should add some new fields which should be populated by validators:
+1. `val_peer_count_low`, `val_peer_count_high`, `val_peer_count_target`. These are the lower bound, upper bound, and target number of validator peers to maintain, respectively.
+2. `isValidator` - whether the node should join the valp2p network.
```

Comment:
I'm not sure if this is meant to be exposed to the operator, because the casing is different than the other proposed fields. The node already knows if it is a validator or not (`git grep 'This node is a validator'`)